python/ray/runtime_context.py

```diff
@@ -0,0 +1,64 @@
+import ray.worker
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class RuntimeContext(object):
+    """A class used for getting runtime context."""
+
+    def __init__(self, worker):
+        assert worker is not None
+        self.worker = worker
+
+    @property
+    def current_job_id(self):
+        """Get current job ID for this worker or driver.
+
+        Returns:
+            If called by a driver, this returns the job ID. If called in
+                a task, return the job ID of the associated driver.
+        """
+        return self.worker.current_job_id
+
+    @property
+    def get_job_configs(self):
+        # TODO: make a job table
+        key = "_JOB_CONFIG_{}".format(self.current_driver_id.hex())
+        result = ray.state.state._execute_command_on_primary("GET", key)
+        if result is None:
+            raise Exception("Can not get job conig.")
+        return json.loads(result, encoding="UTF-8")
+
+    @property
+    def current_actor_id(self):
+        """Get the current actor ID in this worker.
+
+        Returns:
+            The current driver id in this worker.
+        """
+        # only worker mode has actor_id
+        if self.worker.mode != ray.worker.WORKER_MODE:
+            raise Exception("This method should only be called in worker mode")
+        return self.worker.actor_id
+
+    @property
+    def was_current_actor_reconstructed(self):
```

Comment:
Why do we need this btw?

Reply:
In streaming framework, we need this to check whether it's the first time the actor is created.